src/documents/management/commands/document_consumer.py

```diff
@@ -40,14 +41,34 @@ def _tags_from_path(filepath) -> set[Tag]:
     """
     tag_ids = set()
     path_parts = Path(filepath).relative_to(settings.CONSUMPTION_DIR).parent.parts
-    for part in path_parts:
+    for index, part in enumerate(path_parts):
+        # If first subdir should be interpreted as owner
+        # this subdir should not be added as a tag
+        if index == 0 and settings.CONSUMER_SUBDIR_AS_OWNER:
+            owner_id = User.objects.get(username__iexact=part).pk
+            if owner_id:
+                continue
         tag_ids.add(
             Tag.objects.get_or_create(name__iexact=part, defaults={"name": part})[0].pk,
         )
 
     return tag_ids
 
 
+def _owner_from_path(filepath) -> int:
+    """
+    Check first subfolder from filepath below CONSUMPTION_DIR,
+    check if subfolder is equals to an existing user and return user id.
+
+    Returns Owner ID or None
+    """
+    owner_id = None
+    path_parts = Path(filepath).relative_to(settings.CONSUMPTION_DIR).parent.parts
+    owner_id = User.objects.get(username__iexact=path_parts[0]).pk
```

Comment:
If the subfolder name doesn't exist, this raises `model.DoesNotExist`.  Consumption continues, but it's messy.  You should check [`exists`](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#django.db.models.query.QuerySet.exists) first